### DISCUSSION TITLE
plotter.update_scalars will not update the scalar bar

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [pyvista/pyvista#3779] [Q&A]
When I tried to use the method `plotter.update_scalars` to update the scalars that I tried to present. I found the scalar bar will not update automatically. Although I can remove the scalar bar and add a new one and set the new range, I think it's better to add a parameter to `plotter.update_scalars` so that users can choose whether to update the scalar bar or not according to the new scalar data array.
Here is a test code to help to make things clearer. I think the `update_scalars` method is not what I expected when I tried to use it.
```
import pyvista as pv
import numpy as np

mesh = pv.Sphere()
mesh.point_data['data1'] = np.arange(mesh.n_points)
mesh.point_data['data2'] = 10*np.arange(mesh.n_points)

pl = pv.Plotter()
pl.add_mesh(mesh, scalars='data1', show_scalar_bar=False)
pl.update_scalars(scalars='data2')
pl.show()
```

**Comment by banesullivan** [comment]
Yeah, this is a current limitation of PyVista's scalar bar implementation. `update_scalars()` really should handle updating the scalar bar. With the recent work to add `LookupTable` support and better `Mapper` wrappings this shouldn't be too difficult for us to implement.
